File: mobile/plugins/simSharedDefaults/SimSharedDefaults.m
// Objective-C bridge that exposes the Swift SimSharedDefaults class to the
// React Native bridge. RCT_EXTERN_MODULE registers the class by name (the
// @objc(SimSharedDefaults) in the .swift), so no -Swift.h import is needed.
//
// Selectors match the JS calls in deviceToken.ts:
//   NativeModules.SimSharedDefaults.setItem(suite, key, value)
//   NativeModules.SimSharedDefaults.removeItem(suite, key)
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(SimSharedDefaults, NSObject)

RCT_EXTERN_METHOD(setItem:(NSString *)suite key:(NSString *)key value:(NSString *)value)
RCT_EXTERN_METHOD(removeItem:(NSString *)suite key:(NSString *)key)

@end

File: mobile/plugins/simSharedDefaults/SimSharedDefaults.swift
// SimSharedDefaults — tiny native module the JS calls to mirror the share
// device token into the App Group so the Share Extension can read it while the
// app is closed. Interface matches docs/share-extension-setup.md exactly.
//
// Copied into ios/<app>/ at prebuild by plugins/withSimSharedDefaults.js.
// The JS seam lives in mobile/src/lib/share/deviceToken.ts (NativeModules.SimSharedDefaults).
import Foundation

@objc(SimSharedDefaults)
class SimSharedDefaults: NSObject {
  @objc func setItem(_ suite: String, key: String, value: String) {
    UserDefaults(suiteName: suite)?.set(value, forKey: key)
  }

  @objc func removeItem(_ suite: String, key: String) {
    UserDefaults(suiteName: suite)?.removeObject(forKey: key)
  }

  @objc static func requiresMainQueueSetup() -> Bool { false }
}
